chutney/server/src/main/java/com/chutneytesting/execution/infra/storage/DatabaseExecutionHistoryRepository.java

```diff
@@ -197,8 +197,21 @@ public List<ExecutionSummary> getExecutionsWithStatus(ServerReportStatus status)
     @Override
     @Transactional
     public void deleteExecutions(Set<Long> executionsIds) {
+        Set<Long> campaignExecutionsIds = getCampaignExecutionsWithOneScenarioExecution(executionsIds);
+
         scenarioExecutionsJpaRepository.deleteAllByIdInBatch(executionsIds);
         scenarioExecutionReportJpaRepository.deleteAllById(executionsIds);
+        campaignExecutionJpaRepository.deleteAllByIdInBatch(campaignExecutionsIds);
+    }
+
+    private Set<Long> getCampaignExecutionsWithOneScenarioExecution(Set<Long> executionsIds) {
+        return executionsIds.stream()
+            .map(this::getExecutionSummary)
+            .map(ExecutionSummary::campaignReport)
+            .flatMap(Optional::stream)
+            .filter(campaignExecution -> campaignExecution.scenarioExecutionReports().size() == 1)
```

Comment:
I don't understand why deleting campaign execs with only 1 exec ? Why not when there are more exec ?

Reply:
change name to be clearer :
from getCampaignExecutionsWithOneScenarioExecution
to getCampaignExecutionsWithOnlyOneScenarioExecution

Reply:
change the order of delete to respect child/parent foreignkey :
from : 
        scenarioExecutionsJpaRepository.deleteAllByIdInBatch(executionsIds);
        scenarioExecutionReportJpaRepository.deleteAllById(executionsIds);
        campaignExecutionJpaRepository.deleteAllByIdInBatch(campaignExecutionsIds);
to : 
        campaignExecutionJpaRepository.deleteAllByIdInBatch(campaignExecutionsIds);
        scenarioExecutionReportJpaRepository.deleteAllById(executionsIds);
        scenarioExecutionsJpaRepository.deleteAllByIdInBatch(executionsIds);